### PR TITLE
Add OrioleDB extension documentation

### PR DESCRIPTION
**Pull request by za-arthur**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Add OrioleDB extension documentation.

## What is the current behavior?

No documentation.

### CODE REVIEW

**Comment by olirice** [comment]
@charislam and ideas where a better home for this doc would be? Having it addons makes it sound like something you have to pay for

Screenshot 2024-11-27 at 1 30 48 PM

**Review by olirice — CHANGES_REQUESTED** [review]

File: apps/docs/content/guides/platform/orioledb.mdx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+id: 'orioledb'
+title: 'OrioleDB: a cloud-native storage engine for PostgreSQL'
+description: "A storage extension for PostgreSQL which uses PostgreSQL's pluggable storage system"
+---
+
+The [OrioleDB](https://www.orioledb.com/) Postgres extension provides a drop-in replacement storage engine for the default heap storage method. It is designed to improve Postgres' scalability and performance.
+
+OrioleDB is in active development and currently has [certain limitations](https://www.orioledb.com/docs/usage/getting-started#current-limitations). Currently, only B-tree indexes are supported, so features like pg_vector's HNSW indexes are not yet available. We are actively developing an Index Access Method bridge to unlock support for all index types used with heap storage. In Supabase OrioleDB image the default storage method has been updated to use OrioleDB by default, granting better performance out of the box.
```
[the remark this answers is not in the document]
Comment (olirice) [reply apps/docs/content/guides/platform/orioledb.mdx:9]: move this section after the benchmark paragraph and screenshot pls

File: apps/docs/content/guides/platform/orioledb.mdx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+OrioleDB uses index-organized tables, where table data is stored in the index structure. This design eliminates the need for separate heap storage, reduces overhead and improves lookup performance for primary key queries.
+
+### No buffer mapping
+
+In-memory pages are connected to the storage pages using direct links. This allows OrioleDB to bypass PostgreSQL's shared buffer pool and eliminate the associated complexity and contention in buffer mapping.
+
+### Undo log
+
+Multi-Version Concurrency Control (MVCC) is implemented using UNDO log. The UNDO log stores previous row versions and transaction information, which enables consistent reads while reducing the need for table vacuuming and minimizing bloat in most cases.
```
Comment (olirice) [inline apps/docs/content/guides/platform/orioledb.mdx:27]: is it a reduction in need for vacuuming or is vacuum not needed at all?

File: apps/docs/content/guides/platform/orioledb.mdx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+### Copy-on-write checkpoints
+
+OrioleDB implements copy-on-write checkpoints to persist data efficiently. This approach writes only modified data during a checkpoint, reducing the I/O overhead compared to traditional PostgreSQL checkpointing and allowing row-level WAL logging.
+
+### Decoupled storage and compute
+
+OrioleDB has experimental support for separating the storage and compute layers. This decoupling allows storage to scale independently of compute resources and enables more flexible resource allocation. This architecture is particularly suited for cloud environments, where storage and compute resources are often provisioned separately.
```
Comment (olirice) [inline apps/docs/content/guides/platform/orioledb.mdx:35]: note that we don't offer decoupled storage and compute on the platform yet

File: apps/docs/content/guides/platform/orioledb.mdx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  author text not null,
+  published_at timestamptz not null default CURRENT_TIMESTAMP,
+  views bigint not null,
+  primary key (id)
+);
+```
+
+### Creating indexes
+
+OrioleDB tables always have a primary key. If it wasn't defined explicitly, a hidden primary key is created using the `ctid` column.
```
Comment (olirice) [inline apps/docs/content/guides/platform/orioledb.mdx:65]: even though it was already mentioned, pls add another note here that only btree is supported

**Review by olirice — APPROVED** [review]

**Comment by charislam** [comment]
> @charislam and ideas where a better home for this doc would be? Having it addons makes it sound like something you have to pay for
> 
> Screenshot 2024-11-27 at 1 30 48 PM

Database either under a new subsection or under Overview?

File: apps/docs/content/guides/platform/orioledb.mdx (hunk 2 above)
Comment (za-arthur) [reply apps/docs/content/guides/platform/orioledb.mdx:27]: Indeed, vacuum isn't needed. I'll reword this sentence.

File: apps/docs/content/guides/platform/orioledb.mdx (hunk 3 above)
Comment (za-arthur) [reply apps/docs/content/guides/platform/orioledb.mdx:35]: I'll remove this.

**Comment by olirice** [comment]
Frontend changes are coming tomorrow for how the image selection works so we'll need to update those images before merging

**Review by fsansalvadore — APPROVED** [review]

**Pull request merged**